### DISCUSSION TITLE
Suddenly every time I paste with vim it reformats the entire file with code cleanup. How do I disable this?

### DISCUSSION THREAD

**Discussion by fischp** [JetBrains/ideavim#568] [General]
Suddenly every time I paste with vim in Rider using c++(so in both cpp and h files) it reformats the entire file with code cleanup. How do I disable this?

**Comment by citizenmatt** [comment]
Unfortunately, this is a known issue with Rider and IdeaVim. IdeaVim will set the selection before calling an action to reformat the selection, but due to Rider's async nature, the action is executed before the selection is processed. The issue [VIM-2808](https://youtrack.jetbrains.com/issue/VIM-2808/IdeaVim-line-copy-paste-reformats-the-entire-file-for-some-reason) has a couple of workarounds until the issue is fixed properly.
